### PR TITLE
Don't expose HubConnectionContext on the Hub

### LINKED ISSUE #1369: Reduce the surface area on HubConnectionContext
Today we expose the HubConnectionContext on the Hub itself via Context.Connection. This is also the same connection used by HubLifetimeManagers to communicate directly with the socket via the Output channel. We should decouple these 2 things so that we expose less to the Hub itself (for example, it makes little sense for the hub to be writing to the output directly). 

We'll likely need a smaller object that wraps the connection and exposes:

- Items
- User
- ConnectionId
- UserIdentifier
- Abort/AbortToken
- Maybe IP Address

### PR DESCRIPTION
**Pull request by davidfowl**
- Made HubCallerContext an abstract class
- Made DefaultHubCallerContext that gets data from the HubConnectionContext.
- Removed IP address
- Removed Connection property

Fixes #1369

### CODE REVIEW

File: src/Microsoft.AspNetCore.SignalR.Core/DefaultHubCallerContext.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+        public override string UserIdentifier => _connection.UserIdentifier;
+
+        public override ClaimsPrincipal User => _connection.User;
+
+        public override IDictionary<object, object> Items => _connection.Items;
+
+        public override IFeatureCollection Features => _connection.Features;
+
+        public override CancellationToken ConnectionAbortedToken => _connection.ConnectionAbortedToken;
```
Comment (anurse) [inline src/Microsoft.AspNetCore.SignalR.Core/DefaultHubCallerContext.cs]: It's `RequestAborted` not `RequestAbortedToken`, so this should be `ConnectionAborted`

https://github.com/aspnet/HttpAbstractions/blob/b059bcc426ea7593ceeed2df1b6ce14fd3a626ba/src/Microsoft.AspNetCore.Http/DefaultHttpContext.cs#L158-L162

Comment (anurse) [reply src/Microsoft.AspNetCore.SignalR.Core/DefaultHubCallerContext.cs]: I realize you didn't add this property, but the comment is still relevant and a simple fix ;)

Comment (davidfowl) [reply src/Microsoft.AspNetCore.SignalR.Core/DefaultHubCallerContext.cs]: Will do

**Comment by davidfowl** [comment]
```
[xUnit.net 00:00:14.7830100]     Microsoft.AspNetCore.SignalR.Tests.HubEndpointTests.HubMethodDoesNotSendResultWhenInvocationIsNonBlocking [FAIL]
Failed   Microsoft.AspNetCore.SignalR.Tests.HubEndpointTests.HubMethodDoesNotSendResultWhenInvocationIsNonBlocking
Error Message:
 Assert.Null() Failure
Expected: (null)
Actual:   CloseMessage { Error = null }
Stack Trace:
   at Microsoft.AspNetCore.SignalR.Tests.HubEndpointTests.HubMethodDoesNotSendResultWhenInvocationIsNonBlocking() in /_/test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTests.cs:line 545
```

/cc @JamesNK

**Pull request merged**